core/error/error_macros.h

```diff
@@ -100,8 +100,8 @@ void _err_flush_stdout();
  * Always try to return processable data, so the engine can keep running well.
  * Use the _MSG versions to print a meaningful message to help with debugging.
  *
- * The `((void)0)` no-op statement is used as a trick to force us to put a semicolon after
- * those macros, making them look like proper statements.
+ * The `((void)0)` no-op and `do{...} while(false)` statements are used as a trick to force 
```

Comment:
There's a trailing space on this line. And might be good to add spaces in the quoted pseudo code to follow our convention for readability:
```suggestion
 * The `((void)0)` no-op and `do {...} while (false)` statements are used as a trick to force
```